bugbug/tools/code_review.py

```diff
@@ -117,9 +138,145 @@ def get_review_request_by_id(self, review_id: int) -> ReviewRequest:
 
+    @abstractmethod
+    def get_all_inline_comments(
+        self, comment_filter
+    ) -> Iterable[tuple[int, list[InlineComment]]]:
+        raise NotImplementedError
+
+    def load_patch_by_id(self, diff_id) -> Patch:
+        """Load a patch from local cache if it exists.
+
+        If the patch is not in the local cache it will be requested from the
+        provider and cache it locally.
+
+        Args:
+            diff_id: The ID of the patch.
+
+        Returns:
+            The patch.
+        """
+        try:
+            with open(f"patches/{diff_id}.patch", "r") as f:
+                patch = Patch(f.read())
+        except FileNotFoundError:
+            with open(f"patches/{diff_id}.patch", "w") as f:
+                patch = self.get_patch_by_id(diff_id)
+                f.write(patch.raw_diff)
+
+        return patch
+
+    def get_matching_hunk(
+        self, patched_file: PatchedFile, comment: InlineComment
+    ) -> Hunk:
+        def source_end(hunk: Hunk) -> int:
+            return hunk.source_start + hunk.source_length
+
+        def target_end(hunk: Hunk) -> int:
+            return hunk.target_start + hunk.target_length
+
+        if comment.on_removed_code is None:
+            matching_hunks = [
+                hunk
+                for hunk in patched_file
+                if hunk.target_start <= comment.start_line < target_end(hunk)
+                or hunk.source_start <= comment.start_line < source_end(hunk)
+            ]
+
+            # If there is more than one matching hunk, choose the one where the
+            # line number of the comment corresponds to an added or deleted line. We
+            # prioritize added lines over deleted lines because comments are more
+            # likely to be on added lines than deleted lines.
+            if len(matching_hunks) > 1:
+                for hunk in matching_hunks:
+                    for line in hunk:
+                        if line.is_added and line.target_line_no == comment.start_line:
+                            return hunk
+
+                    for line in hunk:
+                        if (
+                            line.is_removed
+                            and line.source_line_no == comment.start_line
+                        ):
+                            return hunk
+
+            if len(matching_hunks) != 0:
+                return matching_hunks[0]
+
+        elif comment.on_removed_code:
+            for hunk in patched_file:
+                if hunk.source_start <= comment.start_line < source_end(hunk):
+                    return hunk
+
+        else:
+            for hunk in patched_file:
+                if hunk.target_start <= comment.start_line < target_end(hunk):
+                    return hunk
+
+    def retrieve_comments_with_hunks(self):
+        def comment_filter(comment: InlineComment):
+            comment_content = comment.content
+
+            # Ignore very short and very log comments
```

Comment:
`long`

Reply:
Fixed in 133f66ba96570cc3a89cdbc7bf7caf1489f9f2b6